KIPs/kip-113.md

```diff
@@ -0,0 +1,252 @@
+---
+kip: 113
+title: BLS public key registry
+author: Ian (@ian0371), Ollie (@blukat29), Joseph (@jiseongnoh), and Aidan (@aidan-kwon)
+discussions-to: https://github.com/klaytn/kips/issues/113
+status: Draft
+type: Standards Track
+category: Core
+created: 2023-05-22
+---
+
+## Simple Summary
+
+A design of simple BLS12-381 registry.
+
+## Abstract
+
+This standard defines an interface for BLS12-381 registry, which enables nodes or smart contracts to access and view all the public keys of GCs. This is not to replace AddressBook, but rather to supplement it in order to facilitate the use of BLS in various components, including signature-based random (KIP-114).
```

Comment:
```suggestion
This standard defines an interface for BLS12-381 registry, which enables nodes or smart contracts to access and view all the public keys of GC members. This is not to replace AddressBook, but rather to supplement it in order to facilitate the use of BLS in various components, including signature-based random (KIP-114).
```

GC is the most appropriate term that we can use? Is there any reason that we can also store non-GC member's public key?

Reply:
We can use validators, or consensus nodes instead of GC.
There's no reason to store non-GC member key; KIP-113 registry only allows CN nodes registered in AddressBook to register their public key.

Reply:
Then, please use the term GC members rather than GCs. Thanks~

Reply:
This section is removed because it no longer defines how the registry as in system contract will be deployed, and thus not limited to storing public keys of GC members.